lib/xdparser2.ts

```diff
@@ -676,92 +676,104 @@ function parseSplitsFromAnswer(answerWithSplits: string, splitCharacter?: string
+        if (index === str.length - 1) {
+          textSlice += str[index]
+        }
       }
+      continue
+    }
 
-      // Bold is started by two asterisks
-      if (letter === "*" && nextLetter === "*") {
-        mode = "bold"
-        // Jump a letter
-        index++
-        pushText()
-        continue
+    if (mdTokens.includes(slice)) {
+      // tries the find the most recent operator that matches the current one
+      while (stack.length > 0 && stack[stack.length - 1]?.[1] !== slice) {
+        let top = stack.pop()
+        if (top) {
+          textSlice = top[1] + textSlice
+        }
       }
 
-      if (letter === "/" && prevLetter === " ") {
-        mode = "italics"
-        pushText()
-        continue
+      // if there is no matching operator, then this will be the starting operator
+      if (stack.length === 0) {
+        if (textSlice.length > 0) {
+          let recent = components[components.length - 1]
+          if (recent?.[0] === "text") {
+            recent[1] += textSlice
+          } else {
+            components.push(["text", textSlice])
+          }
+        }
+        textSlice = ''
+        stack.push([index - 1, slice])
+      } else {
+        // there is one so pop stack and push to components
+        components.push([getMDType(slice), str.slice(stack[stack.length - 1][0] + 2, index - 1)])
+        stack.pop()
+        textSlice = ''
       }
+      index++
+      continue
+    }
+    textSlice += str[index - 1]
+    if (index === str.length - 1) {
+      textSlice += str[index]
+    }
+  }
 
-      if (letter === "~") {
-        mode = "strike"
-        pushText()
-        continue
-      }
-    } else if (mode === "link") {
-      if (letter === "]") {
-        linkText = token
-        token = ""
-        continue
-      } else if (letter === ")") {
-        components.push(["link", linkText, token.slice(1)])
-        token = ""
-        mode = "text"
-        continue
-      }
-    } else if (mode === "bold") {
-      if (letter === "*" && nextLetter === "*") {
-        mode = "text"
-        components.push(["bold", token])
-        index++
-        token = ""
-        continue
-      }
-    } else if (mode === "italics") {
-      if (letter === "/" || !letter) {
-        mode = "text"
-        components.push(["italics", token])
-        token = ""
-        continue
-      }
-    } else if (mode === "strike") {
-      if (letter === "~") {
-        mode = "text"
-        components.push(["strike", token])
-        token = ""
-        continue
+  if (textSlice.length > 0) {
+    components.push(["text", textSlice])
+  }
+
+  const urlRegex = new RegExp(/\[([a-zA-Z ]*)\]\(([a-zA-Z0-9\/:\.\?\&\%\$\!\@\#\-\+\~\,\= ]*)\)/g)
+  let i = 0
+  while (i < components.length) {
+    const component = components[i]
+    if (component[0] === "text") {
+      const matches = component[1].matchAll(urlRegex)
+      const length = component[1].length
+      component[1].replace(urlRegex, '')
```

Comment:
Is this a replace on a string? If so, this line isn't doing anything? Shouldn't it be:

component[1] = component[1].replace(urlRegex,'')